cpp/include/raft/random/rng.hpp

```diff
@@ -21,21 +21,31 @@
 namespace raft {
 namespace random {
```

Comment:
Are we planning to work the `rng_state` class into this PR? If not, it would be worth capturing some of the follow-on changes and discussion in an issue just so they don't get lost.